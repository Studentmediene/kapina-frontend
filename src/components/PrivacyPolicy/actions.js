import {
  LOAD_PRIVACY_POLICY_PENDING,
  LOAD_PRIVACY_POLICY_SUCCESS,
  LOAD_PRIVACY_POLICY_FAILED,
} from './constants';

export function loadPrivacyPolicy() {
  return {
    type: LOAD_PRIVACY_POLICY_PENDING,
  };
}

export function privacyPolicyLoaded(content) {
  return {
    type: LOAD_PRIVACY_POLICY_SUCCESS,
    content,
  };
}

export function privacyPolicyError() {
  return {
    type: LOAD_PRIVACY_POLICY_FAILED,
  };
}
