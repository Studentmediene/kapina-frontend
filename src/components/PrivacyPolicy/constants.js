export const LOAD_PRIVACY_POLICY_PENDING =
  'app/PrivacyPolicy/LOAD_PRIVACY_POLICY_PENDING';
export const LOAD_PRIVACY_POLICY_SUCCESS =
  'app/PrivacyPolicy/LOAD_PRIVACY_POLICY_SUCCESS';
export const LOAD_PRIVACY_POLICY_FAILED =
  'app/PrivacyPolicy/LOAD_PRIVACY_POLICY_FAILED';
