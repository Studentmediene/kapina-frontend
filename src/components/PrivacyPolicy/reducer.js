import { fromJS } from 'immutable';
import {
  LOAD_PRIVACY_POLICY_PENDING,
  LOAD_PRIVACY_POLICY_SUCCESS,
  LOAD_PRIVACY_POLICY_FAILED,
} from './constants';

const initialState = fromJS({
  privacyPolicy: '',
  loading: false,
  error: false,
});

function privacyPolicyReducer(state = initialState, action) {
  switch (action.type) {
    case LOAD_PRIVACY_POLICY_PENDING:
      return state.set('loading', true).set('error', false);
    case LOAD_PRIVACY_POLICY_SUCCESS:
      return state
        .set('loading', false)
        .set('error', false)
        .set('privacyPolicy', action.privacyPolicy);
    case LOAD_PRIVACY_POLICY_FAILED:
      return state.set('loading', false).set('error', true);
    default:
      return state;
  }
}

export default privacyPolicyReducer;
