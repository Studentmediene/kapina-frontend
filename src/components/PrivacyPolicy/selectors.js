import { createSelector } from 'reselect';

const selectPrivacyPolicyDomain = () => state => state.get('privacyPolicy');

const selectContent = () =>
  createSelector(
    selectPrivacyPolicyDomain(),
    about => about.get('content'),
  );

const selectPrivacyPolicyLoading = () =>
  createSelector(
    selectPrivacyPolicyDomain(),
    about => about.get('loading'),
  );

const selectPrivacyPolicyError = () =>
  createSelector(
    selectPrivacyPolicyDomain(),
    about => about.get('error'),
  );

export { selectContent, selectPrivacyPolicyLoading, selectPrivacyPolicyError };
