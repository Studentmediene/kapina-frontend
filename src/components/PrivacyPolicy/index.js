import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import {
  selectContent,
  selectPrivacyPolicyLoading,
  selectPrivacyPolicyError,
} from './selectors';
import Loader from 'components/Loader';
import { loadPrivacyPolicy } from './actions';

export class PrivacyPolicy extends React.Component {
  static propTypes = {
    content: PropTypes.string,
    loading: PropTypes.bool.isRequired,
    error: PropTypes.bool.isRequired,
  };

  static async getInitialProps(ctx) {
    const { store } = ctx;
    if (!selectContent()(store.getState())) {
      store.dispatch(loadPrivacyPolicy());
    }
    return {};
  }

  render() {
    if (this.props.loading) {
      return <Loader />;
    } else if (this.props.error) {
      return <div>Kunne ikke laste inn siden.</div>;
    } else if (this.props.content) {
      return <div dangerouslySetInnerHTML={{ __html: this.props.content }} />;
    } else {
      return <Loader />;
    }
  }
}

const mapStateToProps = createStructuredSelector({
  content: selectContent(),
  loading: selectPrivacyPolicyLoading(),
  error: selectPrivacyPolicyError(),
});

export default connect(mapStateToProps)(PrivacyPolicy);
