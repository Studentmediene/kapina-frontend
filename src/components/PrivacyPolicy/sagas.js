import { call, put, takeEvery } from 'redux-saga/effects';

import { getGraphQL } from 'utils/api';
import { LOAD_PRIVACY_POLICY_PENDING } from './constants';
import { privacyPolicyLoaded, privacyPolicyError } from './actions';

export function* loadPrivacyPolicy() {
  console.log('fetching privacy policy');
  const query = `query {
    settings {
      privacyPolicy
    }
  }`;
  try {
    const result = yield call(getGraphQL, query);
    yield put(privacyPolicyLoaded(result.data.settings.privacyPolicy));
  } catch (error) {
    yield put(privacyPolicyError());
  }
}

export default [takeEvery(LOAD_PRIVACY_POLICY_PENDING, loadPrivacyPolicy)];
