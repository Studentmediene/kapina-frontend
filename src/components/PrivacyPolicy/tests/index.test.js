import React from 'react';
import { shallow } from 'enzyme';

import PrivacyPolicy from '../';

describe('<PrivacyPolicy />', () => {
  it('renders correctly', () => {
    const tree = shallow(<PrivacyPolicy />);
    expect(tree).toMatchSnapshot();
  });
});
