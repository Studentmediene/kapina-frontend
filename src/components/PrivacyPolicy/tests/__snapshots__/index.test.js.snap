// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivacyPolicy /> renders correctly 1`] = `
<div>
  <h1>
    Personvernerklæring
  </h1>
  <p>
    Sist oppdatert: 12. September 2018
  </p>
  <p>
    Denne personvernerklæringen forklarer hvordan Radio Revolt (radiorevolt.no, Radio Revolt på iOS og Radio Revolt på Android) samler inn, behandler og bruker personopplysninger. Erklæringen inneholder opplysninger du har krav på når det samles inn opplysninger fra nettsted og apper (personopplysningsloven § 19) samt generell informasjon om hvordan vi behandler personopplysninger (personopplysningsloven § 18 første ledd). Radio Revolt har ansvar for å følge personopplysningsloven.
  </p>
  <h2>
    Behandlingsansvarlig
  </h2>
  <p>
    Daglig leder for Studentmediene i Trondheim AS er overordnet behandlingsansvarlig for Radio Revolt. Radioredaktør for Radio Revolt er delegert de daglige oppgavene for behandling av personopplysninger. Delegeringen omfatter kun oppgavene og ikke ansvaret. Radioredaktør kan kontaktes på 
    <a
      href="mailto:redaktor@radiorevolt.no"
    >
      epost
    </a>
    .
  </p>
  <h2>
    Bruk av informasjonskapsler
  </h2>
  <p>
    Informasjonskapsler (cookies) er små tekstfiler som plasseres på din datamaskin når du laster ned en nettside.
  </p>
  <p>
    Radio Revolt benytter ingen informasjonskapsler.
  </p>
  <h2>
    Personopplysninger
  </h2>
  <p>
    Radio Revolt lagrer ingen personopplysninger.
  </p>
</div>
`;
